src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -129,8 +129,11 @@ internal void DependencyRegen(WorkspaceModel ws)
             // If package is set to uninstall state, update the package info
             foreach (var package in dependencyViewExtension.pmExtension.PackageLoader.LocalPackages.Where(x => x.MarkedForUninstall))
             {
-                (packageDependencies.Where(x => x.Name == package.Name).FirstOrDefault() as PackageDependencyInfo).State = PackageDependencyState.RequiresRestart;
-                this.RestartBanner.Visibility = Visibility.Visible;
+                if (packageDependencies.Count() != 0)
```

Comment:
How about moving this `if` statement outside of the `for` loop so that if it isn't true, you can straightaway skip the `for` loop.